org-code-javabuilder/lib/src/main/java/org/code/neighborhood/Painter.java

```diff
@@ -0,0 +1,146 @@
+package org.code.neighborhood;
+
+public class Painter {
+  private static int lastId = 0;
+  private int xLocation;
+  private int yLocation;
+  private Direction direction;
+  private int remainingPaint;
+  private Grid grid;
+  private String id;
+
+  public Painter(int x, int y, String direction, int paint) {
+    this.xLocation = x;
+    this.yLocation = y;
+    this.direction = Direction.fromString(direction);
+    if (this.direction == null) {
+      throw new UnsupportedOperationException("Invalid direction given to painter");
```

Comment:
Separate out exceptions that are directly caused by an error in the student's code versus exceptions caused by levelbuilders or by internal server errors. 

Example: If there's an issue with the grid.txt, we'd want to tell users to contact support and provide them with the level ID. If it's an issue with their code, we'd want to provide them with a stack trace.

For now, we don't have the full pipeline set up, but we need different exception types for "caused by the user" versus "caused by the levelbuilder" and possibly for "caused by the system"

Reply:
This one is a student code error if they give an invalid direction but I can update all of the grid reading ones to throw IOExceptions instead (although I think you mentioned that at some point students will be able to define their own grid.txt)

Reply:
True. We can definitely keep the different error messages (see the enum comment below), but creating different exception types will give us better control over how we handle the exceptions and communicate them to the user.